src/pages/contextsAndReducer/CartManagementProvider.jsx

```diff
@@ -0,0 +1,60 @@
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useReducer,
+  useState,
+} from "react";
+
+const CartManagementContext = createContext();
+
+const useCartManager = () => useContext(CartManagementContext);
+
+function CartManagementProvider({ children }) {
+  function reducer(state, action) {
+    switch (action.type) {
+      case "ADD_TO_WISHLIST": {
+        return { ...state, wishlist: action.payload };
+      }
+
+      case "REMOVE_FROM_WISHLIST": {
+        return {
+          ...state,
+          wishlist: action.payload,
+        };
+      }
+
+      case "ADD_TO_CART": {
+        return { ...state, cart: action.payload };
+      }
+
+      case "REMOVE_FROM_CART": {
+        return { ...state, cart: action.payload };
+      }
+
+      case "INCREASE_QUANTITY": {
+        return { ...state, cart: action.payload };
+      }
+
+      case "DECREASE_QUANTITY": {
+        return { ...state, cart: action.payload };
```

Comment:
suggestion - the return value of all these cases are the same, so you can replace them with a single case "SET_CART"